src/sparsebase/preprocess/preprocess.cc

```diff
@@ -415,6 +415,44 @@ IDType *RCMReorder<IDType, NNZType, ValueType>::GetReorderCSR(
   return Q;
 }
 
+#ifdef USE_AMD_ORDER
+#ifdef __cplusplus
+extern "C" {
+#endif
+#include <amd.h>
+#ifdef __cplusplus
+}
+#endif
+
```

Comment:
A default constructor must be implemented. AMDReorderCSR function needs to be implemented in the default constructor as well.